apple/internal/aspects/resource_aspect.bzl

```
@@ -256,9 +256,9 @@ def _apple_resource_aspect_impl(target, ctx):
             ),
         )
 
-    # Get the providers from dependencies, referenced by deps and locations for resources.
+    # Get the providers from dependencies, referenced by deps, impl_deps and locations for resources.
     inherited_providers = []
-    provider_deps = ["deps"] + collect_args.get("res_attrs", [])
+    provider_deps = ["deps"] + ["implementation_deps"] + collect_args.get("res_attrs", [])
```

Any reason to not simplify this list addition?
```suggestion
    provider_deps = ["deps", "implementation_deps"] + collect_args.get("res_attrs", [])
```